test/test_functionalization.py

```diff
@@ -1542,6 +1542,35 @@ def forward(self, x_1):
     add = torch.ops.aten.add.Tensor(view_1, 1);  view_1 = None
     return add""")
 
+    def test_python_functionalization_zero_tensor(self):
+        def f(x):
+            y = torch.ops.aten._efficientzerotensor([4])
+            return x * y
+        x = torch.randn(4)
+        out_ref = f(x)
+        out_test = dispatch_functionalize(f)(x)
+        self.assertEqual(out_ref, out_test)
+
+    def test_python_functionalization_conj(self):
+        def f(x):
+            y = x.conj()
+            return torch.view_as_real(y.resolve_conj())
```

Comment:
Btw, C++ functionalization looks silently incorrect on .conj():
```
import torch
from torch._functorch.aot_autograd import from_fun, to_fun
import torch.utils._pytree as pytree
from functorch import make_fx

def cpp_functionalize(f):
    def inner(*args):
        fun_args = pytree.tree_map_only(torch.Tensor, to_fun, args)
        try:
            torch._enable_functionalization(reapply_views=False)
            fun_outs = f(*fun_args)
        finally:
            torch._disable_functionalization()
        outs = pytree.tree_map_only(torch.Tensor, from_fun, fun_outs)
        return outs
    return inner

def f(x):
    y = x.conj()
    return y

x = torch.randn(4)
gm = make_fx(cpp_functionalize(f))(x)
print(gm.code)
```
What does Python functionalization show for this case? The correct thing here seems like there should have been a call to .conj_physical.

Reply:
Hmm... what about that output makes it silently wrong?

In my head, the relationship between `.conj() <-> .conj_physical()` is analogous to `.transpose() <-> transpose_copy()`. The first toggles some metadata on the tensor saying "this tensor is a view", while the second materializes that view - and only later, once we hit an op that needs the view to be materialized do we go and materialize it (e.g. a kernel that can't handle non-contiguous or conjugated views).

Reply:
I guess one difference is that if we consider AOTDispatcher to be really ahead-of-time tracing out dispatcher functionalities, and `conj -> conj_physical` is implemented as a dispatcher functionality, then you're saying that we expect inductor to never have to worry about handling `conj`, and only have to worry about handling `conj_physical`?

Reply:
Sorry, forgot to paste the output:
```
def forward(self, arg0_1):
    return arg0_1
```
That's what gets traced. Since .conj is a view operation, I would expect that there be a call to conj_physical in the graph instead of no call at all